cwms_radar_api/src/main/java/cwms/radar/data/dao/LocationsDaoImpl.java

```diff
@@ -280,18 +281,18 @@ public Catalog getLocationCatalog(String cursor, int pageSize, String unitSystem
 
         condition = condition.and(AV_LOC2.AV_LOC2.LOCATION_ID.upper().greaterThan(locCursor));
 
-        SelectLimitPercentStep<Record1<Long>> forLimit = dsl.select(AV_LOC2.AV_LOC2.LOCATION_CODE)
+        String locCodeField = "LOCATION_CODE_COL";
+        SelectLimitPercentStep<Record1<Long>> forLimit = dsl.select(AV_LOC2.AV_LOC2.LOCATION_CODE.as(locCodeField))
```

Comment:
forLimit should have an "as name".

Reply:
Then again I might be wrong here. worth printing the query to see what JOOQ does, that might only be for actual tables.

Reply:
I didn't see one on `forLimit` itself. The generated query looks like:

```
select "CWMS_20"."AV_LOC2".*
from "CWMS_20"."AV_LOC2"
where (
                  "CWMS_20"."AV_LOC2"."UNIT_SYSTEM" = 'SI'
              and (regexp_like("CWMS_20"."AV_LOC2"."LOCATION_ID", '.*', 'i'))
              and (regexp_like("CWMS_20"."AV_LOC2"."DB_OFFICE_ID", 'SWT', 'i'))
              and upper("CWMS_20"."AV_LOC2"."LOCATION_ID") > '*'
              and "CWMS_20"."AV_LOC2"."LOCATION_CODE" in (
              select "v0" "LOCATION_CODE_COL"
              from (
                       select
                           "x"."v0",
                           rownum "rn"
                       from (
                                select "CWMS_20"."AV_LOC2"."LOCATION_CODE" "v0"
                                from "CWMS_20"."AV_LOC2" "LOCATION_CODES"
                                where (
                                                  "CWMS_20"."AV_LOC2"."UNIT_SYSTEM" = 'SI'
                                              and (regexp_like("CWMS_20"."AV_LOC2"."LOCATION_ID", '.*', 'i'))
                                              and (regexp_like("CWMS_20"."AV_LOC2"."DB_OFFICE_ID", 'SWT', 'i'))
                                              and upper("CWMS_20"."AV_LOC2"."LOCATION_ID") > '*'
                                              and "CWMS_20"."AV_LOC2"."ALIASED_ITEM" is null
                                          )
                                order by "CWMS_20"."AV_LOC2"."LOCATION_ID"
                            ) "x"
                       where rownum <= (0 + 1000)
                   )
              where "rn" > 0
          )
          )
```

Looks to me like `limit` is expanding in a funky way.

Reply:
I just think that is how jooq handles limit and offset for oracle.

See the oracle example in this https://www.jooq.org/doc/latest/manual/sql-building/sql-statements/select-statement/limit-clause/

keyset pagination is in someways much better but we'd have to redo how the cursors are generated
https://blog.jooq.org/faster-sql-paging-with-jooq-using-the-seek-method/
I did this briefly for the abstract ratings before it got thrown out.

Reply:
The tl;dr version of offset vs keyset pagination is that instead of skipping a certain number of rows (offset) you order the rows by a bunch of columns and you pass around the last values of those columns that you saw in the previous page as the cursor.   Jooq made a "seek" function to make writing the queries easier.  The columns to sort by and the values in the seek clause are coupled.  seek is supposed to be more efficient as the number of pages grows b/c the db doesn't have to count the precise number of rows to skip - you may end up needing to add indexes to the sort columns but that's the sort of thing dba's get paid to figure out -right?

DSL.using(configuration)
   .select(T.ID, T.VALUE)
   .from(T)
   .orderBy(T.VALUE, T.ID)
   .seek(2, 533)
   .limit(5);